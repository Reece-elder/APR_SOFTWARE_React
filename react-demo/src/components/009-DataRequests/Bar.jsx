import { useState, useEffect } from "react";
import axios from 'axios';

import Beer from "./Beer";

const Bar = () => {

    // Data being pulled from API
    const [beerList, setBeerList] = useState([]);

    // Error state
    const [error, setError] = useState(null);

    // Check if Loaded state
    const [loaded, setLoaded] = useState(false)

    // Load time for setLoaded function
    const loadTime = 3000;

    // Use Effect Hook

    // axios.*request type*(url)
    useEffect(() => {
        getData();
    },);


    const getData = () => {
        setTimeout(() => {
            axios.get('https://api.punkapi.com/v2/beers')
            .then((response) => {
                // Setting loaded state to true because data is loaded
                setLoaded(true);
    
                // Set beerList to response to populate array
                setBeerList(response.data)
            })
            .catch((error) => {
                setLoaded(true);
                setError(error);
            });
        }, loadTime);
    }

    const beerBody = {
        name: "beer123",
        image_url: "www.randomimage.com",
        abv: 98,
        food_pairing : {
            0: "apple",
            1: "crisps",
            2: "eggplant"
        }
    };

    const postData = (id) => {
        axios.post(`https://api.punkapi.com/v2/beers/${id}`,beerBody, {
            headers: {
                'Access-Control-Allow-Origin' : '*'
            }
        })
        .then((response) => {
            console.log(response);
        })
        .catch((error) => {
            console.log(error)
            setError(error)
        });
    }

    // Name, image url, abv and food pairing 

    if (error) {
        return <p> Oops, something has gone wrong! {error.message} </p>
    } else if (!loaded) {
        return <p> Please wait, we are loading your data :)</p>
    } else {
        return (
            <div>
                {beerList.map((beer) => (
                    <Beer key={beer.id} name={beer.name} image={beer.image_url} abv={beer.abv} foodPairing={beer.food_pairing}/>
                ))}
            </div>
        )
    }

}
 
export default Bar;