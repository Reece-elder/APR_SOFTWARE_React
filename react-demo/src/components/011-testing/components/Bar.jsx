import Beer from './Beer'

const Bar = () => {

    const name = 'Buzz';
    const image = 'https://images.punkapi.com/v2/keg.png'
    const abv = 4.5;
    const foodPairing = ["food pairing 1", "food pairing 2"];

    return ( 
        <Beer name={name} image={image} abv={abv} foodPairing={foodPairing}/>
     );
}
 
export default Bar;