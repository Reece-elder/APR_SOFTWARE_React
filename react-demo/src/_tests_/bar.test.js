import Bar from "../components/011-testing/components/Bar";

import { create } from "react-test-renderer";

describe(`Bar snapshot test`, () => {

    it(`Should match the snapshot`, () => {
        const element = create(<Bar/>).toJSON();
        expect(element).toMatchSnapshot();
    })

})