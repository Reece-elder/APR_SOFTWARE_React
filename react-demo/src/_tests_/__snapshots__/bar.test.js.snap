// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar snapshot test Should match the snapshot 1`] = `
Array [
  <h2>
    This should break it (hopefully)
  </h2>,
  <div>
    <h3>
      Buzz
    </h3>
    <img
      alt="Buzz"
      src="https://images.punkapi.com/v2/keg.png"
      width={70}
    />
    <p>
      Alcohol content of beer is 
      4.5
    </p>
    <h4>
      Good food pairings with this are: 
    </h4>
    <p>
      food pairing 1
    </p>
    <p>
      food pairing 2
    </p>
  </div>,
]
`;
